SubsoniciousKit/SubsoniciousKit/Authentication/AuthenticationManager.swift

```diff
@@ -12,19 +12,53 @@ import Foundation
 public class AuthenticationManager: ObservableObject {
 
     private let authenticationService: AuthenticationService
+    private let serverPersistenceManager: ServerPersistenceManager
     @Published public private(set) var result: Result<SubsonicResponse, Error>?
+    private var server: Server?
     private var cancellables: Set<AnyCancellable> = []
 
-    public init(authenticationService: AuthenticationService) {
+    public init(authenticationService: AuthenticationService,
+                serverPerstistenceManager: ServerPersistenceManager) {
         self.authenticationService = authenticationService
+        self.serverPersistenceManager = serverPerstistenceManager
+
+        bindResult()
+    }
+
+    private func bindResult() {
+        $result
+            .filter {
+                if case .success = $0 {
+                    return true
+                }
+                return false
+            }
+            .sink { [weak self] _ in
+                guard let server = self?.server else { return }
+                try? self?.serverPersistenceManager.persist(server: server)
+            }
+            .store(in: &cancellables)
     }
 
     public func authenticate(with server: Server) {
+        self.server = server
+
         authenticationService.fetchAuthentication(with: server)
             .map { Result<SubsonicResponse, Error>.success($0) }
             .catch { Just(Result<SubsonicResponse, Error>.failure($0)).eraseToAnyPublisher() }
             .receive(on: DispatchQueue.main)
             .assign(to: \.result, on: self)
             .store(in: &cancellables)
     }
+
+    /// Authenticate with the persisted server if exists.
+    /// - Throws: Error while trying to get the persisted server.
+    /// - Returns: `true` if the persisted server exists; otherwise `false`.
+    public func authenticateWithPersistedServer() throws -> Bool {
+        if let persistedServer = try serverPersistenceManager.persistedServer() {
+            authenticate(with: persistedServer)
+            return true
+        }
+        return false
```

Comment:
why not throw an error instead of return bool ?

Reply:
The first time the user uses the app there is no persisted server. In my opinion, that's not an error.
What do you think?

Reply:
I think It is an error from the code perspective. Ok, it is not a functional error, but you can catch it differently in that case. I think this could be better than a boolean. the name of the method doesn't show why there is a boolean and this choice is drivent by the call place, not your method itself :)

Reply:
Actually my approach was not correct. I'm correcting it :)

Reply:
Should be better now :)